### PR TITLE
task: Add Junit test cases for MySQL SSH tunnel feature

### LINKED ISSUE #26254: MySQL SSH tunnel: Add Junit TC

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Add JUnit test cases for MySQL SSH tunnel feature wherever possible. 
- A new class/file is created to segregate all datasource validity test cases for MySQL, for better readability and code management. However, this PR has only added the SSH tunnel related datasource validity test cases to it. Rest of the datasource validity test cases will be moved to from the current `MySQLPluginTest.java` file. Tracking it here. 

#### PR fixes following issue(s)
Fixes #26254 

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
]
- [x] JUnit

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        DatasourceConfiguration datasourceConfiguration = new DatasourceConfiguration();
+
+        /* Set connection method toggle */
+        ArrayList<Property> properties = new ArrayList<>();
+        properties.add(null);
+        properties.add(new Property("Connection method", "SSH"));
+        datasourceConfiguration.setProperties(properties);
+
+        /* Set connection mode */
+        datasourceConfiguration.setConnection(new com.appsmith.external.models.Connection());
```
Comment (sneha122) [inline app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java:36]: @sumitsum Can we import the Connection model and use it as `new Connection()`? Rather than `new com.appsmith.external.models.Connection()` to be consistent across?

File: app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        sshDatasourceConfiguration.getSshProxy().setPort(null);
+        Set<String> validationErrors = pluginExecutor.validateDatasource(sshDatasourceConfiguration);
+        assertTrue(validationErrors.isEmpty(), validationErrors::toString);
+    }
+
+    /* Test error message for empty SSH username */
+    @Test
+    public void testErrorMessageOnEmptySSHUsername() {
+        DatasourceConfiguration sshDatasourceConfiguration = getDatasourceConfigurationWithSSHConnectionMethod();
+        sshDatasourceConfiguration.getSshProxy().setUsername("");
```
Comment (sneha122) [inline app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java:98]: @sumitsum Should we also test for empty password? Assuming that we throw error when password is not provided

Comment (sumitsum) [reply app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java:98]: Hi @sneha122 , this use case has already been covered via these two test cases: 
1. `testValidateDatasourceNullCredentials` in `MySqlPluginTest.java`
2. `testErrorMessageOnEmptySSHKey` in `MySQLDatasourceValidationTest.java`

File: app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-plugins/mysqlPlugin/src/test/java/com/external/plugins/MySQLDatasourceValidationTest.java]: Done.

**Review by sneha122 — APPROVED** [review]
